src/mesido/head_loss_class.py

```diff
@@ -1436,6 +1436,15 @@ def _pipe_head_loss_constraints(
             )
             constraints.append(((dh - head_loss + flow_dir * big_m) / big_m, 0.0, np.inf))
 
+            constraints.append(
+                (
+                    (-dh - head_loss - (1 - flow_dir) * big_m) / big_m,
+                    -np.inf,
+                    0.0,
+                )
+            )
+            constraints.append(((dh - head_loss - flow_dir * big_m) / big_m, -np.inf, 0.0))
```

Comment:
The current implementation looks like an inequality constraint <= 0.0 when flow_dir = 1. To make this an equality constraint for dh and head_loss taking into account flow_dir is the following constraint not maybe missing:
( (-dh - head_loss + (1 - flow_dir) * big_m) / ...., 0.0, np.inf )

Reply:
That constraint is a few lines above 1430-1436